### DISCUSSION TITLE
does fastdds support psm interface?

### DISCUSSION THREAD

**Discussion by laser-pen** [eProsima/Fast-DDS#2648] [Q&A]
i've noticed that the interface of fastdds is different from the interface of omg psm interface. i wonder whther fastdds support psm interface right now?

**Comment by JLBuenoLopez** [comment]
Hi @laser-pen,

PSM interface is not yet supported by Fast DDS. Currently, Fast DDS only supports PIM API (these discussions also answered this question: #2439 and #2442)

**Reply by skysky97** [reply]
Hi @JLBuenoLopez-eProsima,

Is there any plan for PSM interface?

**Reply by JLBuenoLopez** [reply]
Hi @skysky97,

The PSM API is in our [long term roadmap](https://github.com/eProsima/Fast-DDS/blob/master/roadmap.md#long-term) under the full DDS API compliance. Consequently, do not expect any official release any time soon. You may want to contact our [support team](https://github.com/eProsima/Fast-DDS#getting-help) in case you need to move the PSM interface forward in the roadmap.
